src/framework/application.js

```diff
@@ -1420,8 +1420,8 @@ class Application extends EventHandler {
      * events) so that the canvas resolution is immediately updated.
      */
     updateCanvasSize() {
-        // Don't update if we are in VR
-        if ((this.vr && this.vr.display) || (this.xr.active && this.xr.type === XRTYPE_VR)) {
+        // Don't update if we are in VR or XR
+        if ((!this._allowResize) || (this.xr.active)) {
```

Comment:
Matches logic for WebVR in resizeCanvas: https://github.com/playcanvas/engine/blob/e76b5faa9c5f34ba9a0dbde424596aaa2cfa6fd8/src/framework/application.js#L1376